Source/WebKit/UIProcess/API/glib/WebKitClipboardPermissionRequest.cpp

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2023 Igalia S.L.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ * You should have received a copy of the GNU Library General Public License
+ * along with this library; see the file COPYING.LIB.  If not, write to
+ * the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "config.h"
+#include "WebKitClipboardPermissionRequest.h"
+
+#include "WebKitClipboardPermissionRequestPrivate.h"
+#include "WebKitPermissionRequest.h"
+#include <wtf/glib/WTFGType.h>
+
+#if !ENABLE(2022_GLIB_API)
+typedef WebKitPermissionRequestIface WebKitPermissionRequestInterface;
+#endif
+
+/**
+ * WebKitClipboardPermissionRequest:
+ * @See_also: #WebKitPermissionRequest, #WebKitWebView
+ *
+ * A permission request for reading clipboard contents.
```

Comment:
We'll never want to have a permission request for writing to the clipboard, then? That will just always be allowed no matter what?

Reply:
I don't know, but for now reading is the only operation requiring permission. We can change it later if needed.